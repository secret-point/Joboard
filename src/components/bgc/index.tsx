/* eslint-disable react/jsx-wrap-multilines */
import React from "react";
import ReactDOM from "react-dom";

const Index = () => {
  return <div>Index</div>;
};

// export default index;

ReactDOM.render(<Index />, document.getElementById("root"));
