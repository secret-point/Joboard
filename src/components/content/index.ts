export { default } from "./content";
export { default as Content } from "./content";