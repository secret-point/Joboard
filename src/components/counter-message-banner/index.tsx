import React, { useState, useEffect } from "react";
import {
  MessageBanner,
  MessageBannerType
} from "@amzn/stencil-react-components/message-banner";
import ICandidateApplication from "../../@types/ICandidateApplication";
import isEmpty from "lodash/isEmpty";
import moment from "moment";

const HOUR_IN_SECONDS = 3600;
const MIN_IN_SECONDS = 60;
const RESERVED_TIME_IN_HOURS = 3;

interface CounterMessageBannerProps {
  application: ICandidateApplication;
  onUpdatePageId: Function;
}

let myInterval: any = null;
let reserveTime: number;

const CounterMessageBanner: React.FC<CounterMessageBannerProps> = ({
  application,
  onUpdatePageId
}) => {
  const [minutes, setMinutes] = useState(0);
  const [hours, setHours] = useState(0);

  useEffect(() => {
    if (!isEmpty(application)) {
      const startCountTime: string = application.jobSelected?.jobSelectedOn;
      reserveTime =
        moment(startCountTime).unix() +
        HOUR_IN_SECONDS * RESERVED_TIME_IN_HOURS -
        moment().unix();
      setHours(Math.floor(reserveTime / HOUR_IN_SECONDS));
      setMinutes(
        Math.floor((reserveTime - hours * HOUR_IN_SECONDS) / MIN_IN_SECONDS)
      );
      if (reserveTime < MIN_IN_SECONDS) {
        onUpdatePageId("session-timeout");
      }
    }

    if (reserveTime > MIN_IN_SECONDS) {
      myInterval = setInterval(() => {
        setHours(Math.floor(reserveTime / HOUR_IN_SECONDS));
        setMinutes(
          Math.floor((reserveTime - hours * HOUR_IN_SECONDS) / MIN_IN_SECONDS)
        );
        if (reserveTime < MIN_IN_SECONDS) {
          console.log("timeout");
          clearInterval(myInterval);
        }
      }, MIN_IN_SECONDS * 1000);
    }
  }, [hours, application, minutes, onUpdatePageId]);

  return (
    <MessageBanner type={MessageBannerType.Warning}>
      {`Spot reserved for ${hours} hours and ${minutes} minutes`}
    </MessageBanner>
  );
};

export default CounterMessageBanner;
