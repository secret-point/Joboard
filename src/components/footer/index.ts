export { default } from "./footer";
export { default as Footer } from "./footer";