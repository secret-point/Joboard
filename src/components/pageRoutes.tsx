export enum PAGE_ROUTES {
  DEFAULT = "/",
  CONSENT = "consent",
  PRE_CONSENT = "pre-consent",
  JOB_OPPORTUNITIES = "job-opportunities",
  JOB_CONFIRMATION = "job-opportunities/job-confirmation",
  JOB_DESCRIPTION = "job-opportunities/job-description",
  CONTINGENT_OFFER = "contingent-offer",
  BACKGROUND_CHECK = "bgc",
  BACKGROUND_CHECK_FCRA = "bgc/fcra",
  RESUME_APPLICATION = "resume-application",
  NHE = "nhe",
  ALREADY_APPLIED = "already-applied",
  SELF_IDENTIFICATION = "self-identification",
  REVIEW_SUBMIT = "review-submit",
  THANK_YOU = "thank-you",
  SESSION_TIMEOUT = "session-timeout",
  WOTC = "wotc",
  WOTC_COMPLETE = "wotc-complete",
  ASSESSMENT = "assessment",
  ASSESSMENT_CONSENT = "assessment-consent",
  ASSESSMENT_NOT_ELIGIBLE = "assessment-not-eligible",
  ASSESSMENT_FINISHED = "assessment-finished",
  CANDIDATE_WITHDRAWS = "candidate-withdraws",
  CALI_DISCLOSURE = "cali-disclosure",
  AMAZON_REJECTS = "amazon-rejects",
  WORKFLOW_FAILED = "workflow-failed",
  AMAZON_WITHDRAWS = "amazon-withdraws",
  NO_AVAILABLE_TIME_SLOTS = "no-available-time-slots",
  CAN_NOT_OFFER_JOB = "can-not-offer-job",
  SUPPLEMENTARY_SUCCESS = "supplementary-success",
  REHIRE_ELIGIBILITY_STATUS = "rehire-eligibility-status",
  NO_AVAILABLE_SHIFT = "no-available-shift",
  APPLICATIONID_NULL = "applicationId-null",
  SHIFT_PREFERENCE = "job-opportunities/shift-preferences",
  ADDITIONAL_INFORMATION = "additional-information"
}

// Reference: https://code.amazon.com/packages/HVHStepFunctionDefinitions/blobs/ad75b9869a017bc84a9e03920e835a6b2cec7063/--/configuration/nacf_tier_1_node_workflow/defintion.json#L186
export const PagesControlledByWorkFlowService = [
  PAGE_ROUTES.JOB_OPPORTUNITIES,
  PAGE_ROUTES.ASSESSMENT_CONSENT,
  PAGE_ROUTES.ASSESSMENT,
  PAGE_ROUTES.CONTINGENT_OFFER,
  PAGE_ROUTES.BACKGROUND_CHECK,
  PAGE_ROUTES.CAN_NOT_OFFER_JOB,
  PAGE_ROUTES.CALI_DISCLOSURE,
  PAGE_ROUTES.NHE,
  PAGE_ROUTES.SELF_IDENTIFICATION,
  PAGE_ROUTES.REVIEW_SUBMIT,
  PAGE_ROUTES.THANK_YOU,
  PAGE_ROUTES.WOTC,
  PAGE_ROUTES.SUPPLEMENTARY_SUCCESS,
  PAGE_ROUTES.ASSESSMENT_NOT_ELIGIBLE,
  PAGE_ROUTES.REHIRE_ELIGIBILITY_STATUS,
  PAGE_ROUTES.AMAZON_WITHDRAWS,
  PAGE_ROUTES.CANDIDATE_WITHDRAWS,
  PAGE_ROUTES.AMAZON_REJECTS,
  PAGE_ROUTES.WORKFLOW_FAILED,
  PAGE_ROUTES.RESUME_APPLICATION,
  PAGE_ROUTES.SHIFT_PREFERENCE,
  PAGE_ROUTES.ADDITIONAL_INFORMATION
];
