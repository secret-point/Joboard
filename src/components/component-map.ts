import Loader from "./loader";
import {
  TextBox,
  Button,
  PageTitle,
  PhoneNumber,
  RedirectButton,
  Checkbox,
  DisqualifiedQuestions,
  Text,
  PageHeader,
  Steps,
  DetailedCheckbox,
  Link,
  DetailedRadioButton,
  DetailedRadioButtonList,
  Action,
  Markdown,
  UL,
  StatusLink,
  Dropdown,
  DatePicker,
  MessageBanner,
  Card,
  Image
} from "@amzn/hvh-candidate-application-ui-components/lib";
import PanelContainer from "../containers/panel/panel-container";
import FlyoutContainer from "../containers/flyout";
import ImageContainer from "./image";
import { Col, Row, Spacer, View } from "@stencil-react/components/layout";
import { Card as StencilCard } from "@stencil-react/components/card";

const ComponentMap: any = {
  loader: Loader,
  TextBox,
  Button,
  PageTitle,
  PhoneNumber,
  RedirectButton,
  Checkbox,
  DisqualifiedQuestions,
  Text,
  PageHeader,
  Steps,
  DetailedCheckbox,
  StatusLink,
  Link,
  DetailedRadioButton,
  DetailedRadioButtonList,
  Action,
  Markdown,
  UL,
  Dropdown,
  DatePicker,
  MessageBanner,
  Card,
  Panel: PanelContainer,
  Flyout: FlyoutContainer,
  Image,
  ImageContainer,
  Col,
  Row,
  Spacer,
  View,
  StencilCard
};

export default ComponentMap;
