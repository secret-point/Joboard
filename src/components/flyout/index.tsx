import React from "react";
import {
  FlyoutButtonBar,
  FlyoutContainer,
  FlyoutContent,
  FlyoutHeader,
  FlyoutTitle,
  FlyoutCloseButton,
  WithFlyout
} from "@stencil-react/components/flyout";
import { Row } from "@stencil-react/components/layout";
import { Text } from "@stencil-react/components/text";
import { Button } from "@stencil-react/components/button";
import RendererContainer from "../../containers/renderer";
import Icon from "../icon";

interface IFlyOut {
  flyOut: any;
  onButtonClick: Function;
}

const FlyOut: React.FC<IFlyOut> = ({ flyOut, onButtonClick }) => {
  const renderFlyOut = ({ close }: { close: any }) => {
    const submit = () => {
      if (flyOut.submitButtonProps.actionProps) {
        const { actionProps } = flyOut.submitButtonProps;
        onButtonClick(actionProps.action, { options: actionProps.options });
      }
      close();
    };
    const cancel = () => {
      if (flyOut.closeButtonProps.actionProps) {
        const { actionProps } = flyOut.closeButtonProps;
        onButtonClick(actionProps.action, { options: actionProps.options });
      }
      close();
    };
    return (
      <FlyoutContainer>
        <FlyoutHeader alignItems="center" justifyContent="space-between">
          <FlyoutTitle>{flyOut.title}</FlyoutTitle>
          <FlyoutCloseButton onClick={close} />
        </FlyoutHeader>
        <FlyoutContent>
          <RendererContainer
            isConfigContainsFlyouts={true}
            flyOutId={flyOut.id}
            type={flyOut.placeholder || "content"}
            flyOutConfig={flyOut}
          />
        </FlyoutContent>
        <FlyoutButtonBar justifyContent="flex-end">
          {flyOut.closeButtonProps && (
            <Button onClick={cancel} tertiary>
              {flyOut.closeButtonProps.label || "Cancel"}
            </Button>
          )}
          {flyOut.submitButtonProps && (
            <Button onClick={submit} primary>
              {flyOut.submitButtonProps.label || "Submit"}
            </Button>
          )}
        </FlyoutButtonBar>
      </FlyoutContainer>
    );
  };
  return (
    <WithFlyout flyoutPosition={flyOut.position} renderFlyout={renderFlyOut}>
      {({ openFlyout }) => (
        <Button onClick={openFlyout} {...flyOut.buttonProps}>
          <Row gridGap={8} alignItems="center">
            {flyOut.buttonIconProps &&
              flyOut.buttonIconProps?.position === "left" && (
                <Icon icon={flyOut.buttonIconProps.icon} />
              )}
            <Text {...flyOut.buttonProps?.style}>{flyOut.buttonLabel}</Text>
            {flyOut.buttonIconProps &&
              flyOut.buttonIconProps?.position === "right" && (
                <Icon icon={flyOut.buttonIconProps.icon} />
              )}
          </Row>
        </Button>
      )}
    </WithFlyout>
  );
};

export default FlyOut;
