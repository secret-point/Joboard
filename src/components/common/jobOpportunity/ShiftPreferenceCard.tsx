import React from "react";
import { Col, Row } from "@amzn/stencil-react-components/layout";
import { H4, Text } from "@amzn/stencil-react-components/text";
import { IconArrowRight, IconSize } from "@amzn/stencil-react-components/icons";
import { translate as t } from "../../../utils/translator";
import { routeToAppPageWithPath } from "../../../utils/helper";
import { PAGE_ROUTES } from "../../pageRoutes";

const ShiftPreferenceCard = () => {
  return (
    <Col
      className="schedulePreferenceCard"
      padding="S300"
      gridGap={8}
    >
      <H4>
        {t("BB-ShiftPreferenceCard-title-text", "Don't see a shift that works for you?")}
      </H4>
      <Text fontSize="T200">
        {t("BB-ShiftPreferenceCard-content-text", "You can submit your shift preferences and we'll contact you when we find a shift that works for you.")}
      </Text>
      <Row padding={{ top: "S400" }}>
        <Row
          className="shiftPreferenceLink"
          gridGap={5}
          onClick={() => {
            routeToAppPageWithPath(PAGE_ROUTES.SHIFT_PREFERENCE);
          }}
        >
          <Text fontWeight="medium" fontSize="T200">
            {t("BB-JobOpportunity-Go-To-shift-preference-Link", "Select your shift preferences")}
          </Text>
          <IconArrowRight size={IconSize.Medium} fontSize="T100" aria-hidden />
        </Row>
      </Row>
    </Col>
  );
};

export default ShiftPreferenceCard;