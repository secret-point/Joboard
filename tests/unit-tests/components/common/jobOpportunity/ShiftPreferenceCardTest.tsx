import { shallow } from "enzyme";
import React from "react";
import ShiftPreferenceCard from "../../../../../src/components/common/jobOpportunity/ShiftPreferenceCard";
import { PAGE_ROUTES } from "../../../../../src/components/pageRoutes";
import { mountWithStencil } from "@amzn/stencil-react-components/tests";

describe("ShiftPreference Card", () => {
  it("should match snapshot", () => {
    const shallowWrapper = shallow(
      <ShiftPreferenceCard />);

    expect(shallowWrapper).toMatchSnapshot();
  });

  it("it should open shift preference on click", () => {
    window.location.hash = `#/${PAGE_ROUTES.JOB_OPPORTUNITIES}`;
    const wrapper = mountWithStencil(<ShiftPreferenceCard />);
    const goBackToShiftSelectionButton = wrapper.find("Row.shiftPreferenceLink");
    goBackToShiftSelectionButton.simulate("click");

    wrapper.update();
    expect(window.location.hash).toEqual(`#/${PAGE_ROUTES.SHIFT_PREFERENCE}`);
  });
});