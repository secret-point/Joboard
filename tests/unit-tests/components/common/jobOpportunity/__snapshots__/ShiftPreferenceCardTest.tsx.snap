// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShiftPreference Card should match snapshot 1`] = `
<Col
  className="schedulePreferenceCard"
  gridGap={8}
  padding="S300"
>
  <H4
    color="neutral90"
    font="primary"
    fontSize="T400"
    fontWeight="regular"
    hyphens="none"
    margin={0}
  >
    Don't see a shift that works for you?
  </H4>
  <Text
    font="primary"
    fontSize="T200"
    hyphens="none"
  >
    You can submit your shift preferences and we'll contact you when we find a shift that works for you.
  </Text>
  <Row
    padding={
      Object {
        "top": "S400",
      }
    }
  >
    <Row
      className="shiftPreferenceLink"
      gridGap={5}
      onClick={[Function]}
    >
      <Text
        font="primary"
        fontSize="T200"
        fontWeight="medium"
        hyphens="none"
      >
        Select your shift preferences
      </Text>
      <IconArrowRight
        aria-hidden={true}
        display="block"
        fontSize="T100"
        size={24}
      />
    </Row>
  </Row>
</Col>
`;
